astroquery/cadc/core.py

```diff
@@ -609,7 +609,7 @@ def exec_sync(self, query, maxrec=None, uploads=None, output_file=None,
             if isinstance(output_file, str):
                 fname = output_file
             elif hasattr(output_file, 'name'):
-                fname = output_file.name
+                fname = output_file
```

Comment:
If `output_file` is a `Patħ` then the following should work also with older `astropy` versions.
```suggestion
                fname = str(output_file)
```

Reply:
OK, let's try this. This case wasn't tested at all, so I don't know what other cases should be tested, Path is one of the most obvious ones that should work.